XCUITests/SyncFAUITests.swift

```diff
@@ -0,0 +1,77 @@
+        XCTAssertTrue(app.webViews.buttons["Sign in"].exists)
+    }
+
+    func testPlaceholderValues () {
+        navigator.goto(FxASigninScreen)
+        waitforExistence(app.webViews.staticTexts["Sign in"])
+        let mailPlaceholder = "Email"
+        let passwordPlaceholder = "Password"
+
+        let defaultMailPlaceholder = app.webViews.textFields["Email"].placeholderValue!
+        let defaultPasswordPlaceholder = app.webViews.secureTextFields["Password"].placeholderValue!
+        XCTAssertTrue(mailPlaceholder == defaultMailPlaceholder, "The mail placeholder does not show the correct value")
+        XCTAssertTrue(passwordPlaceholder == defaultPasswordPlaceholder, "The password placeholder does not show the correct value")
+    }
+
+    func testTypeOnGivenFields() {
+        navigator.goto(FxASigninScreen)
+        waitforExistence(app.webViews.staticTexts["Sign in"])
+
+        // Tap Sign in without any value in email Password focus on Email
+        navigator.performAction(Action.TapOnSignInButton)
+        waitforExistence(app.webViews.staticTexts["Valid email required"])
+
+        // Enter only email, wrong and correct and tap sign in
+        navigator.performAction(Action.TypeInvalidEmail)
+        navigator.performAction(Action.TapOnSignInButton)
+        waitforExistence(app.webViews.staticTexts["Valid email required"])
+
+        navigator.performAction(Action.TypeCorrectEmail)
+        navigator.performAction(Action.TapOnSignInButton)
+        waitforExistence(app.webViews.staticTexts["Valid password required"])
+
+        // Enter invalid (too short) and incorrect password
+        navigator.performAction(Action.TypeInvalidPassword)
+        navigator.performAction(Action.TapOnSignInButton)
+        waitforExistence(app.webViews.staticTexts["Must be at least 8 characters"])
+
+        navigator.performAction(Action.TypeIncorrectPassword)
+        navigator.performAction(Action.TapOnSignInButton)
+        waitforExistence(app.webViews.staticTexts["Unknown account."])
+        XCTAssertTrue(app.webViews.links["Sign up"].exists)
+    }
+
+    func testCreateAnAccountLink() {
+        navigator.goto(FxASigninScreen)
+        waitforExistence(app.webViews.links["Create an account"])
+        navigator.goto(FxCreateAccount)
+        waitforExistence(app.webViews.buttons["Create account"])
+    }
+
+    func testShowPassword() {
+        navigator.goto(FxASigninScreen)
+        waitforNoExistence(app.webViews.staticTexts["Show"])
+        // Typing on Email should not show Show option
+        navigator.performAction(Action.TypeCorrectEmail)
+        waitforNoExistence(app.webViews.staticTexts["Show"])
+        // Typing on Password should show Show option
+        navigator.performAction(Action.TypeInvalidPassword)
+        waitforExistence(app.webViews.staticTexts["Show"])
+        // After removing password Show option should not be shown
+        app.keys["delete"].press(forDuration: 2)
```

Comment:
What does this do? Could you leave a comment that explains this line?

Reply:
Added a comment in the test explaining what it does and tried to better articulate the comments above too. Hope it is clearer now.
Thanks!